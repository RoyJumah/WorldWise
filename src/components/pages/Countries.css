.selected-country-info {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.fade-enter {
  opacity: 0;
}

.fade-enter-active {
  opacity: 1;
  transition: opacity 300ms;
}

.fade-exit {
  opacity: 1;
}

.fade-exit-active {
  opacity: 0;
  transition: opacity 300ms;
}

.selected-country-flag {
  width: 200px;
  height: 150px;
  margin-bottom: 20px;
}

.selected-country-container {
  display: flex;
}
