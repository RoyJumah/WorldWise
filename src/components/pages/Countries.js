import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';

import { motion } from 'framer-motion';

import './Countries.css';

const Countries = () => {
  const { countries } = useSelector((state) => state.countries);

  const alpha3Code = useParams().alpha3Code || '';

  const selectedCountry = countries.find(
    (country) => country.alpha3Code === alpha3Code,
  );

  if (!selectedCountry) {
    return <p>Country not found</p>;
  }

  const listVariants = {
    hidden: { opacity: 0 },
    visible: { opacity: 1, transition: { staggerChildren: 0.1 } },
  };

  const listItemVariants = {
    hidden: { opacity: 0, y: 20 },
    visible: { opacity: 1, y: 0 },
  };

  return (
    <motion.div
      initial={{ opacity: 0, y: 30 }}
      animate={{ opacity: 1, y: 0 }}
      exit={{ opacity: 0, y: -30 }}
      transition={{ duration: 0.7 }}
    >
      <div className="selected-country-info">
        <div className="title__name">
          <img
            src={selectedCountry.flag}
            alt={selectedCountry.name}
            className="selected-country-flag"
          />
          <h2>{selectedCountry.name}</h2>
        </div>
      </div>
      <div className="selected-country-details">
        <h3 className="country-details">Stats Breakdown</h3>
      </div>
      <motion.ul
        className="selected-country-details-content"
        variants={listVariants}
        initial="hidden"
        animate="visible"
      >
        <motion.li variants={listItemVariants}>
          <strong>Capital:</strong>
          {' '}
          {selectedCountry.capital}
        </motion.li>
        <motion.li variants={listItemVariants}>
          <strong>Timezones:</strong>
          {' '}
          {selectedCountry.timezones}
        </motion.li>
        <motion.li variants={listItemVariants}>
          <strong>Population:</strong>
          {' '}
          {selectedCountry.population.toLocaleString()}
        </motion.li>
        <motion.li variants={listItemVariants}>
          <strong>Area:</strong>
          {' '}
          {selectedCountry.area.toLocaleString()}
          {' '}
          km
          <sup>2</sup>
        </motion.li>
        <motion.li variants={listItemVariants}>
          <strong>Languages:</strong>
          {' '}
          {selectedCountry.languages
            .map((language) => language.name)
            .join(', ')}
        </motion.li>
        <motion.li variants={listItemVariants}>
          <strong>Currencies:</strong>
          {' '}
          {selectedCountry.currencies
            .map((currency) => currency.name)
            .join(', ')}
        </motion.li>
        {selectedCountry && selectedCountry.borders ? (
          <motion.li variants={listItemVariants}>
            <strong>Border Countries:</strong>
            {selectedCountry.borders.length > 0 ? (
              selectedCountry.borders.join(', ')
            ) : (
              <em>None</em>
            )}
          </motion.li>
        ) : null}
      </motion.ul>
    </motion.div>
  );
};

export default Countries;
