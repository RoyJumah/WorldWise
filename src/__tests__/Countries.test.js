import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import store from '../redux/ConfigureStore';
import Countries from '../components/pages/Countries';

import filterCountries from '../tertiary_test_files/countries-list';

describe('Display Countries component', () => {
  test('it matches snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Route path="/countries/:alpha3Code?">
              <Countries />
            </Route>
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should return  Countries Component', () => {
    render(
      <Provider store={store}>
        <Router>
          <Route path="/countries/:alpha3Code?">
            <Countries />
          </Route>
        </Router>
      </Provider>,
    );
  });
});

describe('filterCountries', () => {
  test('it filters the countries based on search term', () => {
    const countries = [
      { name: 'United States of America' },
      { name: 'United Kingdom' },
      { name: 'France' },
    ];
    const searchTerm = 'United';
    const filteredCountries = filterCountries(countries, searchTerm);
    expect(filteredCountries).toEqual([
      { name: 'United States of America' },
      { name: 'United Kingdom' },
    ]);
  });
});
