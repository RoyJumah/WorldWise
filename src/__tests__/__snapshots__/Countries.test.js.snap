// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Countries component it matches snapshot 1`] = `null`;
